sklearn/utils/fixes.py

```diff
@@ -335,3 +335,19 @@ def __call__(self, *args, **keywords):
             return self.func(*args, **kwargs)
 else:
     from functools import partial
+
+
+if np_version < (1, 6, 2):
+    # Allow bincount to accept empty arrays
+    # https://github.com/numpy/numpy/commit/40f0844846a9d7665616b142407a3d74cb65a040
+    def bincount(x, weights=None, minlength=None):
+        if len(x) > 0:
+            return np.bincount(x, weights, minlength)
+        else:
+            if minlength is None:
+                minlength = 0
+            minlength = np.asscalar(np.asarray(minlength, dtype=np.intp))
```

Comment:
I guess `int(minlength)` will accept strings like `'123'` as arguments which is not the case of the `minlength` argument in `np.bincount`.

From looking at the C code + some playing around, my understanding is that minlength is allowed to be an int or a scalar array with an int (int or anything castable to an int actually).

Full disclosure with my current code passing `minlenth='123'` into the function will result in an error not on the line
```python
minlength = np.asscalar(np.asarray(minlength, dtype=np.intp))
```
after which `minlength` is still equal to `'123'` but on the next line because `np.zeros` needs an int or sequence of ints as its first argument.